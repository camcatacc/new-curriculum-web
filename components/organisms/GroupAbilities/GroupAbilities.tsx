// Modules
import React from "react";

// Elements
import Ability from "components/molecules/Ability/Ability";

// Definitions
import { CmsAbility } from "interfaces/cms/molecules/Ability";

export interface GroupAbilitiesProps {
	abilities: CmsAbility[];
}

// Element
const GroupAbilities = ({ abilities }: GroupAbilitiesProps) => {
	return (
		<div className="grid md:gap-x-12 gap-y-10 grid-cols-2 md:grid-cols-3 gap-x-4">
			{abilities.map((ab, ind) => (
				<Ability description={ab.text} title={ab.title} icon={ab.icon} key={`ab${ind}`} />
			))}
		</div>
	);
};

export default GroupAbilities;
