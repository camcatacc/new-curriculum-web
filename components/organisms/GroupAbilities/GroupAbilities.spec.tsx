// Modules
import React from "react";
import { render } from "@testing-library/react";

// Elements
import GroupAbilities from "components/organisms/GroupAbilities/GroupAbilities";
import { ImportContactsSharp } from "@material-ui/icons";

// Definitions
import type { GroupAbilitiesProps } from "components/organisms/GroupAbilities/GroupAbilities";

// Default Props
const defaultProps: GroupAbilitiesProps = {
	abilities: [
		{ description: "This ability is the best", icon: ImportContactsSharp, title: "Best ability" },
		{ description: "This ability is the 2 best", icon: ImportContactsSharp, title: "Best ability 2" }
	]
};

// Tests
describe("GroupAbilities", () => {
	it("Renders as expected", () => {
		const { container } = render(<GroupAbilities {...defaultProps} />);
		expect(container).toMatchSnapshot();
	});
});
