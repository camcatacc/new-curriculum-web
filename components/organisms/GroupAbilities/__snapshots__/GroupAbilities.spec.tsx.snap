// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupAbilities Renders as expected 1`] = `
<div>
  <div
    class="grid md:gap-x-12 gap-y-10 grid-cols-2 md:grid-cols-3 gap-x-4"
  >
    <div
      class="w-full h-full flex-col flex items-center"
    >
      <div
        class="md:w-1/5 w-2/5"
      >
        <div
          class="sc-gsDKAQ faXWdV rounded-full w-min h-min"
          height="100%"
          width="100%"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root sc-dkPtRN kNoqAZ"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              color="#7986cb"
              d="M17.218,2.268L2.477,8.388C2.13,8.535,2.164,9.05,2.542,9.134L9.33,10.67l1.535,6.787c0.083,0.377,0.602,0.415,0.745,0.065l6.123-14.74C17.866,2.46,17.539,2.134,17.218,2.268 M3.92,8.641l11.772-4.89L9.535,9.909L3.92,8.641z M11.358,16.078l-1.268-5.613l6.157-6.157L11.358,16.078z"
            />
          </svg>
        </div>
      </div>
      <div
        class="mt-4 mb-2"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter"
        >
          Best ability 1
        </h3>
      </div>
      <div
        class=""
      >
        <p
          class="MuiTypography-root sc-bdvvtL gedMyf MuiTypography-body2 MuiTypography-alignCenter"
        >
          This text is 
          important
        </p>
      </div>
    </div>
    <div
      class="w-full h-full flex-col flex items-center"
    >
      <div
        class="md:w-1/5 w-2/5"
      >
        <div
          class="sc-gsDKAQ faXWdV rounded-full w-min h-min"
          height="100%"
          width="100%"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root sc-dkPtRN kNoqAZ"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              color="#7986cb"
              d="M17.218,2.268L2.477,8.388C2.13,8.535,2.164,9.05,2.542,9.134L9.33,10.67l1.535,6.787c0.083,0.377,0.602,0.415,0.745,0.065l6.123-14.74C17.866,2.46,17.539,2.134,17.218,2.268 M3.92,8.641l11.772-4.89L9.535,9.909L3.92,8.641z M11.358,16.078l-1.268-5.613l6.157-6.157L11.358,16.078z"
            />
          </svg>
        </div>
      </div>
      <div
        class="mt-4 mb-2"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter"
        >
          Best ability 2
        </h3>
      </div>
      <div
        class=""
      >
        <p
          class="MuiTypography-root sc-bdvvtL gedMyf MuiTypography-body2 MuiTypography-alignCenter"
        >
          This text is 
          important
        </p>
      </div>
    </div>
  </div>
</div>
`;
